src/Core/Framework/Migration/MigrationException.php

```diff
@@ -18,6 +20,25 @@ class MigrationException extends HttpException
     final public const FRAMEWORK_MIGRATION_INVALID_VERSION_SELECTION_MODE = 'FRAMEWORK__MIGRATION_INVALID_VERSION_SELECTION_MODE';
     final public const FRAMEWORK_MIGRATION_INVALID_MIGRATION_SOURCE = 'FRAMEWORK__INVALID_MIGRATION_SOURCE';
     final public const FRAMEWORK_MIGRATION_IMPLAUSIBLE_CREATION_TIMESTAMP = 'FRAMEWORK__MIGRATION_IMPLAUSIBLE_CREATION_TIMESTAMP';
+    final public const FRAMEWORK_MIGRATION_PLUGIN_COULD_NOT_BE_FOUND = 'FRAMEWORK__MIGRATION_PLUGIN_COULD_NOT_BE_FOUND';
+    final public const FRAMEWORK_MIGRATION_MORE_THAN_ONE_PLUGIN_FOUND = 'FRAMEWORK__MIGRATION_MORE_THAN_ONE_PLUGIN_FOUND';
+    final public const FRAMEWORK_MIGRATION_DIRECTORY_COULD_NOT_BE_CREATED = 'FRAMEWORK__MIGRATION_DIRECTORY_COULD_NOT_BE_CREATED';
+
+    /**
+     * @deprecated tag:v6.7.0 - reason:return-type-change - Will only return `self` in the future
+     */
+    public static function invalidArgument(string $message): self|\InvalidArgumentException
+    {
+        if (!Feature::isActive('v6.7.0.0')) {
+            return new \InvalidArgumentException($message);
+        }
+
+        return new self(
+            Response::HTTP_INTERNAL_SERVER_ERROR,
+            AdapterException::INVALID_ARGUMENT,
```

Comment:
that needs to be a constant in this class. you cannot reference other domain exceptions